packages/react/src/components/form/form.tsx

```diff
@@ -11,9 +11,10 @@ import { useConfiguration } from "../../hooks/use-configuration";
 
 export type Props = {
   className?: string;
+  onSuccess?: () => void;
 };
 
-export const Form: React.FC<Props> = ({ className }) => {
+export const Form: React.FC<Props> = ({ className, onSuccess }) => {
```

Comment:
Do you think we could pass the `onSuccess` callback to `flowState` instead? This is a matter of flow control, rather than UI rendering. In that case `onSuccess` could be called internally, as soon as login is done? If a customer wants to perform redirect on success, that would happen without `Success` being rendered briefly.

Reply:
Also we'd have a guarantee that the callback was called only once. If we involve `useEffect` it gets tricky :)